hypothesis-python/tests/cover/test_stateful.py

```diff
@@ -958,3 +959,17 @@ class Foo(RuleBasedStateMachine):
         targets=(), function=lambda self: 2, arguments={}
     )
     assert len(Foo.initialize_rules()) == 2
+
+
+def test_steps_printed_despite_BaseException(capsys):
```

Comment:
I think given that we've established we don't want this to happen for all base exceptions we should probably have a test that asserts that this *doesn't* happen for e.g. `KeyboardInterrupt` too.

Reply:
Yep, and that it doesn't happen for `pytest.skip()`!